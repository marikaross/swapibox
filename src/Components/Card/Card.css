article {
  background-color: #221f20;
  border: 1px solid #56C0D4;
  color: #56C0D4;
  font-weight: 200;
  width: 20%;
  height: 30%;
  border-radius: 12 px;
  margin: 10px;
  font-size: 1em; }
