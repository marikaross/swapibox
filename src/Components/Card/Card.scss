.peopleCard,
.planetCard,
.vehicleCard {
  background-color: 
}