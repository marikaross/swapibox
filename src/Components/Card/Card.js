import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';

const Card = ({name, homeworld, species, population, terrain, climate, residents, model, starship_class, passengers}) => {
  const peopleCard =
    <article className='peopleCard card' key={name}>
      <h2>{name}</h2>
      <h3>homeworld: {homeworld}</h3>
      <h3>species: {species}</h3>
      <h3>population: {population}</h3>
      <button className='favorites'></button>
    </article>
  
  const planetCard = 
    <article className='planetCard card' key={name}>
      <h2>{name}</h2>
      <h3>terrain: {terrain}</h3>
      <h3>climate: {climate}</h3>
      <h3>population: {population}</h3>
      <h3>residents: {residents}</h3>
      <button className='favorites'></button>
    </article>

  const vehicleCard = 
    <article className='vehicleCard card' key={name}>
      <h2>{name}</h2>
      <h3>model: {model}</h3>
      <h3>class: {starship_class}</h3>
      <h3># of passengers: {passengers}</h3>
      <button className='favorites'></button>
    </article>


  if (homeworld) {
    return peopleCard
  } else if (terrain) {
    return planetCard
  } else {
    return vehicleCard
  }
};


export default Card;


Card.propTypes = {
  name: PropTypes.string,
  homeworld: PropTypes.string,
  species: PropTypes.string,
  population: PropTypes.number,
  terrain: PropTypes.string,
  climate: PropTypes.string,
  residents: PropTypes.string,
  model: PropTypes.string,
  type: PropTypes.string,
  passengers: PropTypes.number
};