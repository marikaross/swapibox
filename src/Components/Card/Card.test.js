import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card.js';

describe('Card', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const mockCard = [{name: 'Bingo', 
      homeworld: 'farm', 
      species: 'canine', 
      population: 15 }]

    const wrapper = shallow(<Card />)

    expect(wrapper).toMatchSnapshot()
  } )

  it('only render planet cards when there is a terrain prop', () => {
    const mockCard = [{name: 'Earth', terrain: 'varied', 
      climate: 'varied', 
      population: 30000000000,
      residents: 'Me. I am the most important'}]
    const wrapper = shallow(<Card card={card} />)
    expect(wrapper.find('.planetCard').length).toEqual(10)
  })

})