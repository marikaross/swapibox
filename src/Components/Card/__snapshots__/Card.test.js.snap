// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        <h3>
          model: 
</h3>,
        <h3>
          class: 
</h3>,
        <h3>
          # of passengers: 
</h3>,
        <button
          className="favorites"
/>,
      ],
      "className": "vehicleCard card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "model: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "model: ",
          undefined,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "class: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "class: ",
          undefined,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "# of passengers: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "# of passengers: ",
          undefined,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "favorites",
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <h3>
            model: 
</h3>,
          <h3>
            class: 
</h3>,
          <h3>
            # of passengers: 
</h3>,
          <button
            className="favorites"
/>,
        ],
        "className": "vehicleCard card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "model: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "model: ",
            undefined,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "class: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "class: ",
            undefined,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "# of passengers: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "# of passengers: ",
            undefined,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "favorites",
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
